### DISCUSSION TITLE
interfacing with Carla

### DISCUSSION THREAD

**Discussion by coderofsalvation** [falkTX/Carla#1453] [Q&A]
Hi,

Is there a way to connect to a carla instance (OSC?) and receive the audiodata (floats) for a particular note/channel?

REASON: I'd like to add plugin-support for an opensource (samplebased) tracker software, but I'd rather interface with superior plugin-hosting software like Carla, instead of implementing/maintaining some kind of inferior pluginhost.
Also, on many platforms the 'easiest' way is by setting up virtual soundcards & mididevices etc, which is pretty non-straightforward compared to 'hey just run Carla if you want pluginsupport'.

**Comment by falkTX** [comment]
You can do this for sure.
Carla current exposes its rack-mode and internal patchbay-mode as plugins, for which you can have full control.

See https://github.com/falkTX/kuriborosu/blob/main/src/kuriborosu.c for example of Carla's API being used in another project, with `carla_get_native_rack_plugin` being the start point.
Once you get a host handle you can use the complete API as documented here https://kx.studio/ns/dev-docs/CarlaBackend/group__CarlaHostAPI.html

PS: I am converting this ticket as a "discussion" since it is more suitable that way
